src/nspawn/nspawn.c

```diff
@@ -378,6 +379,33 @@ static void parse_share_ns_env(const char *name, unsigned long ns_flag) {
         arg_clone_ns_flags = (arg_clone_ns_flags & ~ns_flag) | (r > 0 ? 0 : ns_flag);
 }
 
+static void parse_mount_settings_env(void) {
+        int r;
+        const char *e;
+
+        e = getenv("SYSTEMD_NSPAWN_API_VFS_WRITABLE");
+        if (!e)
+                return;
+
+        if (streq(e, "network")) {
+                arg_mount_settings |= MOUNT_APPLY_APIVFS_RO;
+                arg_mount_settings |= MOUNT_APPLY_APIVFS_NETNS;
+                return;
+        }
+
+        r = parse_boolean(e);
+
+        if (r < 0) {
+                log_warning_errno(r, "Failed to parse SYSTEMD_NSPAWN_API_VFS_WRITABLE from environment, ignoring.");
+                return;
+        } else if (r > 0) {
+                arg_mount_settings &= ~MOUNT_APPLY_APIVFS_RO;
+                arg_mount_settings &= ~MOUNT_APPLY_APIVFS_NETNS;
+        } else {
+                arg_mount_settings |= MOUNT_APPLY_APIVFS_RO;
```

Comment:
Hmm, both of you have a point ;-)

1. Given that `SYSTEMD_NSPAWN_API_VFS_WRITABLE` will mainly be used by machines (i.e. rkt) it is a fair argument to say that we'd expect to have a full read-only filesystem from that perspective.

2. From a user perspective I'd expect backwards compat, since as an user I'd assume that `SYSTEMD_NSPAWN_API_VFS_WRITABLE` is false internally anyways.

After rethinking I am leaning towards 1. because we then have the control of the target read-write settings at the expense of a possible user surprise.